File: iris/python/irispy/test/test_required_containment.py
import irispy
import numpy as np
import unittest

class ContainmentTest(unittest.TestCase):
    def test_required_containment(self):
        obstacles = [np.array([[0., 1],
                               [0, 0]]),
                     np.array([[1., 1],
                               [0, 1]]),
                     np.array([[1., 0],
                               [1, 1]]),
                     np.array([[0., 0],
                               [1, 0]])]
        required_containment_pts = [np.array([1.5, 1.5])]
        start = np.array([0.1, 0.1])

        region = irispy.inflate_region(obstacles, start,
                                       require_containment=True,
                                       required_containment_points=required_containment_pts)
        self.assertTrue(region.polyhedron.getNumberOfConstraints() == 0, "polyhedron should be empty")

if __name__ == '__main__':
    unittest.main()

File: iris/src/test/test_required_containment.cpp
#include "test_util.h"
#include "iris/iris.h"

int main() {
  // Run IRIS with the required_containment_points including a point which is outside the feasible set, which should result in an empty polyhedron

  iris::IRISProblem problem(2);
  problem.setSeedPoint(Eigen::Vector2d(0.1, 0.1));

  Eigen::MatrixXd obs(2,2);
  // Inflate a region inside a 1x1 box
  obs << 0, 1,
         0, 0;
  problem.addObstacle(obs);
  obs << 1, 1,
         0, 1;
  problem.addObstacle(obs);
  obs << 1, 0,
         1, 1;
  problem.addObstacle(obs);
  obs << 0, 0,
         1, 0;
  problem.addObstacle(obs);

  iris::IRISOptions options;
  options.require_containment = true;
  std::vector<Eigen::VectorXd> required_containment_points = {Eigen::Vector2d(1.5, 1.5)};
  options.required_containment_points = required_containment_points;

  auto region = inflate_region(problem, options);
  if (region.polyhedron.getNumberOfConstraints() > 0) {
    throw std::runtime_error("polyhedron should be empty");
  }

  return 0;
}